### PR TITLE
Fix CAgg on CAgg different column order

### LINKED ISSUE #5131: [Bug]: Time bucket error while creating an CAGG on another one
### What type of bug is this?

Unexpected error

### What subsystems and features are affected?

Continuous aggregate

### What happened?

When trying to create a CAGG on another one, we encountered the following error :   
 `ERROR: time bucket function must reference a hypertable dimension column`

While investigate on the error, we found that the bug was thrown if the order of the time partitioning column mismatch between the first materialized view and the hypertable. For exemple, if the time partitioning column is the second in the table and the `time bucket` column is the first in the CAGG, creating another CAGG will cause this error.

As workaround, we use the same position for the `time bucket`  as the time column on the table ; but we manage to have in some case the time column in the sixth position and only four column in the CAGG :sweat_smile: (recreating the CAGG with empty column did the trick)

### TimescaleDB version affected

2.9.1

### PostgreSQL version used

13

### What operating system did you use?

timescale/timescaledb:2.9.1-pg13

### What installation method did you use?

Docker

### What platform did you run on?

On prem/Self-hosted

### Relevant log output and stack trace

```bash
ERROR:  time bucket function must reference a hypertable dimension column
```

### How can we reproduce the bug?

```bash
CREATE TABLE data(
    temperature INT8,
    date TIMESTAMPTZ NOT NULL
);

SELECT create_hypertable('data', 'date');

CREATE MATERIALIZED VIEW cagg
WITH(timescaledb.continuous) AS
SELECT time_bucket('1hour', date) AS hour,
  min(temperature),  
  avg(temperature),
  max(temperature)
FROM data
GROUP BY hour;

CREATE MATERIALIZED VIEW cagg_on_cagg
WITH(timescaledb.continuous) AS
SELECT time_bucket('1day', hour) AS day,
  min(min),
  avg(avg),
  max(max)
FROM cagg
GROUP BY day;
```

### PR DESCRIPTION
**Pull request by fabriziomello**
Creating a CAgg on a CAgg where the time column is in a different order of the original hypertable was raising the following exception:

`ERROR:  time bucket function must reference a hypertable dimension column`

The problem was during the validation we're initializing internal data structure with the wrong hypertable metadata.

Fixes #5131

### CODE REVIEW

File: tsl/test/expected/cagg_on_cagg.out [hunk 1]
```diff
@@ -18,22 +18,33 @@
...
 Running local hypertable tests
 \endif
 SET ROLE :ROLE_DEFAULT_PERM_USER;
--- CAGGs on CAGGs tests
-CREATE TABLE conditions (
-  time :TIME_DIMENSION_DATATYPE NOT NULL,
-  temperature NUMERIC
-);
+DROP TABLE IF EXISTS conditions CASCADE;
+psql:include/cagg_on_cagg_setup.sql:14: NOTICE:  table "conditions" does not exist, skipping
```
Comment (RafiaSabih) [inline tsl/test/expected/cagg_on_cagg.out:36]: Is this necessary?

Comment (fabriziomello) [reply tsl/test/expected/cagg_on_cagg.out:36]: At this point no but the new file `include/cagg_on_cagg_setup.sql` will be included several times so we'll need to cleanup the hypertable to re-setup the test case

**Review by RafiaSabih — APPROVED** [review]

**Review by akuzm — APPROVED** [review]
Weird that codecov complains about the lines that should be covered...

**Comment by fabriziomello** [comment]
> Weird that codecov complains about the lines that should be covered...

It is really weird. In another simple fix I've added a failing test and codecov complains exactly the code I added to fix the failing test  :-(

**Pull request merged**